src/handlers/opportunitiesHandlers.ts

```diff
@@ -0,0 +1,134 @@
+import { getLogger } from '../logger';
+import { db } from '../database';
+import { isOpportunity, isOpportunityArraySchema } from '../types';
+import { ValidationError } from '../errors';
+import type {
+  Request,
+  Response,
+} from 'express';
+import type {
+  Opportunity,
+  CanonicalField,
+  ApplicationSchema,
+} from '../types';
+import type { ApplicationSchemaField } from '../types/ApplicationSchemaField';
+
+const logger = getLogger(__filename);
+
+const getOpportunities = (req: Request, res: Response): void => {
+  db.sql('opportunities.fetchAll')
+    .then((opportunitiesRaw) => {
+      logger.debug(opportunitiesRaw, 'opportunities.fetchAll');
+      type ShortOpportunity = Omit<Opportunity, 'applicationSchema'>;
+      type ShortApplicationSchema = Omit<ApplicationSchema, 'fields'>;
+      type ShortApplicationSchemaField = Omit<ApplicationSchemaField, 'canonicalField'>;
+
+      // Here is how each raw, de-normalized row fetched from the DB looks:
+      interface RowPerField {
+        opportunity: ShortOpportunity;
+        applicationSchema: ShortApplicationSchema;
+        applicationSchemaField: ShortApplicationSchemaField;
+        canonicalField: CanonicalField;
+      };
+
+      const { rows } = opportunitiesRaw;
+      const formalizeRow = (it: any): RowPerField => {
+        const opportunity: ShortOpportunity = {
```

Comment:
I hear ya -- I *suspect* that this will be moot after refactor (which is not to say it's not a good discussion but maybe we can punt it for later / when it's definitely relevant.)